scipy/special/tests/test_hypergeometric.py

```diff
@@ -25,39 +25,66 @@ def test_nan_inputs(self, a, b, x):
+         (0.0030949064301273865, 1.0, 1e-05, 1.0356696454116199),
+         (0.0030949064301273865, 1.0, 0.00021544346900318823, 1.0261510362481985),
+         (0.0030949064301273865, 1.0, 0.004641588833612777, 1.0166326903402296),
+         (0.0030949064301273865, 1.0, 0.1, 1.0071174207698674),
+         (0.1509924314279033, 1.0, 1e-05, 2.806173846998948),
+         (0.1509924314279033, 1.0, 0.00021544346900318823, 2.3092158526816124),
+         (0.1509924314279033, 1.0, 0.004641588833612777, 1.812905980588048),
+         (0.1509924314279033, 1.0, 0.1, 1.3239738117634872),
+         (-0.010678995342969011, 1.0, 1e-05, 0.8775194903781114),
+         (-0.010678995342969011, 1.0, 0.00021544346900318823, 0.9101008998540128),
+         (-0.010678995342969011, 1.0, 0.004641588833612777, 0.9426854294058609),
+         (-0.010678995342969011, 1.0, 0.1, 0.9753065150174902),
+         (-0.06556622211831487, 1.0, 1e-05, 0.26435429752668904),
+         (-0.06556622211831487, 1.0, 0.00021544346900318823, 0.4574756033875781),
+         (-0.06556622211831487, 1.0, 0.004641588833612777, 0.6507121093358457),
+         (-0.06556622211831487, 1.0, 0.1, 0.8453129788602187),
+         (-0.21628242470175185, 1.0, 1e-05, -1.2318314201114489),
+         (-0.21628242470175185, 1.0, 0.00021544346900318823, -0.6704694233529538),
+         (-0.21628242470175185, 1.0, 0.004641588833612777, -0.10795098653682857),
+         (-0.21628242470175185, 1.0, 0.1, 0.4687227684115524)]
     )
-    def test_gh_15650(self, a, b, x, expected):
+    def test_gh_15650_mp(self, a, b, x, expected):
         # See https://github.com/scipy/scipy/issues/15650
+        # b == 1, |a| < 0.25, 0 < x < 1
         #
-        # Reference values computed with mpmath
+        # This purpose of this test is to check the accuracy of results
+        # in the region that was impacted by gh-15650.
+        #
+        # Reference values computed with mpmath using the script:
         #
         # import itertools as it
         # import numpy as np
         #
         # from mpmath import mp
         #
+        # rng = np.random.default_rng(1234)
+        #
         # cases = []
-        # for a, x in it.product(np.logspace(-15, -1, 4), repeat=2):
+        # for a, x in it.product(
+        #         np.random.uniform(-0.25, 0.25, size=6),
+        #         np.logspace(-5, -1, 4),
+        # ):
         #     with mp.workdps(100):
-        #         cases.append((a, 1.0, x, float(mp.hyperu(a, 1.0, x))))
+        #         cases.append((float(a), 1.0, float(x), float(mp.hyperu(a, 1.0, x))))
         assert_allclose(sc.hyperu(a, b, x), expected, rtol=1e-13)
 
+    def test_gh_15650_sanity(self):
+        # The purpose of this test is to sanity check hyperu in the region that
+        # was impacted by gh-15650 by making sure there are no excessively large
+        # results, as were reported there.
+        a = np.linspace(-0.5, 0.5, 500)
+        x = np.linspace(1e-6, 1e-1, 500)
```

Comment:
Just checking because it looks funny - is this supposed to be logspace? (But based on having worked with this failure a lot, I would agree that linear spacing is probably better here if `a` is linspaced.)

Reply:
Yeah, I intended this to be `linspace`.